src/core/sock/sockinfo_ulp.cpp

```diff
@@ -752,32 +754,25 @@ ssize_t sockinfo_tcp_ops_tls::tx(xlio_tx_call_attr_t &tx_arg)
         }
     }
 
+    uint8_t *iv = is_tx_tls13() ? nullptr : m_tls_info_tx.iv;
+    mem_desc *zc_owner = is_zerocopy ? reinterpret_cast<mem_desc *>(tx_arg.priv.mdesc) : nullptr;
     for (ssize_t i = 0; i < tx_arg.attr.sz_iov; ++i) {
         pos = 0;
         while (pos < p_iov[i].iov_len) {
             tls_record *rec;
             ssize_t ret2;
-            size_t sndbuf = m_p_sock->sndbuf_available();
-            size_t tosend = p_iov[i].iov_len - pos;
+            size_t tosend = std::min<size_t>(p_iov[i].iov_len - pos, TLS_RECORD_MAX);
```

Comment:
Question: Can TLS record have 16K payload or 16K should include the TLS header/trailer?

Reply:
according to RFC 8446 sections 5.1 and 5.2, this is limitation of the plaintext payload (before adding inner record type in TLS1.3). TLS1.2 describes limits similarly.